runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/policy/Acceptors.kt

```diff
@@ -48,9 +49,11 @@ class SuccessAcceptor(state: RetryDirective, val success: Boolean) : Acceptor<An
  * @param state The [RetryDirective] that applies when the acceptor's condition is matched.
  * @param errorType The [KClass] of error for this acceptor.
  */
-class ErrorTypeAcceptor(state: RetryDirective, val errorType: KClass<*>) : Acceptor<Any, Any>(state) {
+class ErrorTypeAcceptor(state: RetryDirective, val errorType: String) : Acceptor<Any, Any>(state) {
     override fun matches(request: Any, result: Result<Any>): Boolean =
-        errorType.isInstance(result.exceptionOrNull())
+        result.isFailure && result.exceptionOrNull()!!.let {
+            it::class.simpleName == errorType || (it as? ServiceException)?.sdkErrorMetadata?.errorCode == errorType
```

Comment:
question/sanity check

When we parse error code do we normalize it? e.g. to `FooError` instead of `aws.protocoltests.restjson#FooError`

I want to say yes based on how we generate exception matching.

Reply:
Hmm I wonder if you couldn't just simplify this entire check to:

```kotlin
result.isFailure && result.exceptionOrNull()!!.let { (it as? ServiceException)?.sdkErrorMetadata?.errorCode == errorType }
```

All operation errors are modeled as a `ServiceException` and should have `errorCode` populated. 

Nothing wrong with checking the class, perhaps it saves us in some edge case where `errorCode` isn't populated in the error metadata (but that also is probably a bug)